lib/rspec/core/output_wrapper.rb

```diff
@@ -3,13 +3,17 @@ module Core
     # @private
     class OutputWrapper
       # @private
-      attr_writer :output
+      attr_accessor :output
 
       # @private
       def initialize(output)
         @output = output
       end
 
+      def respond_to?(name, priv=false)
+        @output.respond_to?(name, priv)
+      end
```

Comment:
`puts` is an `IO` instance method, so it gets defined below, so this class will respond to it w/o overriding `respond_to?`.  Unless I'm missing something, the only behavioral change here is that it makes this class claim to respond to methods that it doesn't.  Methods in `IO.instance_methods(false)` are the only ones we proxy to `@output`, and the class already claims to respond to those because they are defined with `define_method` instead of method missing.  OTOH, extra methods that `@output` responds to (such as `StringIO#string`), which are not part of `IO.instance_methods(false)` are not methods that `OutputWrapper` responds to (since we are not using `method_missing`) but that this `respond_to?` implementation will falsely claim that we respond to.

Does that make sense?